### DISCUSSION TITLE
Request a firmware build - 2.8.1

### DISCUSSION THREAD

**Discussion by pfeerick** [EdgeTX/edgetx#3204] [General]
Ask here if you want/need specific options that are not included in the stock builds. Builds will be uploaded to [here (same link as 'By request' below)](http://files.portabledev.xyz/d/8a6b44ee4ae446db8941/), so check it out before making a request just in case there is already a custom build with your chosen options...

The Language builds are automated builds based off of 2.8.1

*Please remember to specify your transmitter when requesting a build* ;)

- [Language builds](https://github.com/pfeerick/lang-firmwares/releases/tag/release) - zip file per transmitter containing all languages built, with a second zip for builds with `heli` page enabled. 
- [Language builds - more](https://github.com/pfeerick/lang-firmwares/releases/tag/latest) - zip file per transmitter containing all languages built, with variants for BLUETOOTH, HELI, BLUETOOTH+HELI, LUAMIXER where size and validity of options permit
- [By request](http://files.portabledev.xyz/d/8a6b44ee4ae446db8941/)

**Comment by jiri-jirus** [comment]
hello, please, would it be possible to create TX16S EN BLUETOOTH builds automatically as new version of EdgeTX is released? many of us has this radio with BT, so we will always want this build.

Thank you

**Reply by pfeerick** [reply]
Sure can. I've added both bluetooth colorlcd multi-language builds, and also lua_mixer, so will see how they go... usually takes about 3 hours to build the full set, probably a bit more than 4 now 🤭

**Reply by tedsears** [reply]
Thanks Pete!  You are always super nice about building the Bluetooth version, but if it's automated we won't have to bug you each time... Thanks!

**Reply by pfeerick** [reply]
Check out the second link above - now points to where the further variants exist -> https://github.com/pfeerick/lang-firmwares/releases/tag/latest

**Comment by gharris07** [comment]
Can you please post build 2.8.1 with Bluetooth enabled and Heli mode enabled for the Radiomaster TX16S and Radiomaster Boxer also.

**Reply by pfeerick** [reply]
Check out the second link above - now points to where the further variants exist, including BLUETOOTH+HELI -> https://github.com/pfeerick/lang-firmwares/releases/tag/latest

**Comment by Helipro** [comment]
Can you please post a build with the T16 with Mode 2, Bluetooth, Gvars, Us units, External access mod, Lua, and  with NO heli and NO override channel function.  Thanks so much!
EDIT: I was able to compile my own under the 2.8.1 branch with everything working properly just as was done for 2.8.
Thanks again!

**Comment by templeofthelord** [comment]
Hello, would it be possible to integrate a option to reverse the stick direction (throttle, elevator, aileron, rudder stick) for Taranis 9XE Hall?
The directions are shown wrong on the display, especially throttle and elevator.
Thank you

**Reply by pfeerick** [reply]
Is this with the x9e-hall firmware? As there are two - x9e and x9e-hall

**Reply by templeofthelord** [reply]
Hello, thank you for the "speedy" reply.
Firmware is " x9e-hall-c1bc4d5.bin" from the website http://edgetx.org/getedgetx.html
This bug was long time ago also on older OpenTx-Versions, later versions fixed it.
An integrated button for changing the directions of the sticks (for example in the calibration dialog), so that they will be shown correctly on the black/white display were be comfortable. 
It is a pity that the automatic firmware compilier/download on EdgeTx-companion not work like before in OpenTx-companion.

**Reply by templeofthelord** [reply]
I try the file x9e-hall-DE-54e96ee (bin file) the bug is the same, then i go back to opentx  - the stick display is correct and all other is fine

**Reply by pfeerick** [reply]
How are you flashing the firmware onto the radio? As the x9e firmware should be for the non-hall sticks, and the x9e-hall for the hall sticks, which literally just reverses the elevator and throttle direction. Probably worth double-checking on the radio settings version page that it lists 'horussticks' in the options list - as if that's not present then probably running the wrong firmware or there is something wrong with the build.

**Reply by yds** [reply]
@templeofthelord if you have a X9E radio with original stock gimbals then you need to flash the non-Hall fw to your radio.

on the other hand, if you upgraded your radio with M9 gimbals then you must use the X9E-Hall fw.

- OpenTX with `horussticks` option selected == EdgeTX X9E-Hall
- OpenTX with `horussticks` option not selected == EdgeTX X9E

**Reply by pfeerick** [reply]
Can I just clarify also - I just realised you specifically said:
> The directions are shown wrong on the **display**, especially throttle and elevator.

By this, do you mean these sticks themselves move the channels in the correct direction (without adding reversing to the mix or input)?

**Reply by templeofthelord** [reply]
My Radio https://www.mhm-modellbau.de/frsky-taranis-x9e.php

_By this, do you mean these sticks themselves move the channels in the correct direction (without adding reversing to the mix or input)?_
That`s right


    OpenTX with horussticks option selected == EdgeTX X9E-Hall
    OpenTX with horussticks option not selected == EdgeTX X9E

I tried all this 

_How are you flashing the firmware onto the radio?_
flashing from SD-Card

_horussticks' in the options list_
is not shown with both builds, but in the older opentx-version (which is ok) it is also not shown


An integrated button for changing the directions of the sticks is missing

An automatic firmware compilier/download to choose all wanted options (for example shutdownconfirm and others) on EdgeTx-companion like before in OpenTx-companion is missing

So i will wait going to EdgeTx till this options are available, but i am afraid this will never be...

Thank you and greetings from Germany

**Reply by gagarinlg** [reply]
You are wrong there, take a look at
https://github.com/EdgeTX/cloudbuild

We are working on it.

**Reply by pfeerick** [reply]
Are you *sure* about that? 
![image](https://github.com/EdgeTX/edgetx/assets/5500713/55423626-692e-4aa8-a814-6ff83d35e6fa)

**Reply by templeofthelord** [reply]
![grafik](https://github.com/EdgeTX/edgetx/assets/132878766/a8398a57-c73d-4b9a-b067-a725aff0e217)
i will check it again later and then i will give a feedback

**Comment by Rmsf33** [comment]
Hello, could you please edit the Taranis X9D Plus file? I would like it to be in Portuguese and that the Heli function is enabled. Thank you very much!

**Reply by pfeerick** [reply]
I've done two builds - link is in the first post - one is specifically v2.8.1, and the other is the current state of the 2.8 branch, i.e. a preview of 2.8.5. 

`FLAVOR=x9dp EXTRA_OPTIONS="-DTRANSLATIONS=PT -DHELI=ON" tools/build-gh.sh`

**Reply by Rmsf33** [reply]
> Eu fiz duas compilações - o link está no primeiro post - uma é especificamente v2.8.1 e a outra é o estado atual da ramificação 2.8, ou seja, uma prévia de 2.8.5.
> 
> `FLAVOR=x9dp EXTRA_OPTIONS="-DTRANSLATIONS=PT -DHELI=ON" tools/build-gh.sh`

Thank you very much! It worked, I just upgraded the firmware. I just noticed an oscillation in the sticks, if you look at the sticks graphics they are oscillating. But I performed a test and found that the same error occurs in all EdgeTx versions.

**Reply by pfeerick** [reply]
If by "oscillation" you mean the numbers are changing just slightly, then yes, a little noise is normal... the sample rate is that high you get that slight jitter shown on the display. We're probably going to drop showing the digit after the decimal so it's not visible, as it is of no consequence once fed to a flight controller, or dampened out by the jitter filter.
